src/systems/filecoin_mining/storage_mining/storage_miner_actor.go

```diff
@@ -2,7 +2,20 @@ package storage_mining
 
 import base_mining "github.com/filecoin-project/specs/systems/filecoin_mining"
 import sector "github.com/filecoin-project/specs/systems/filecoin_mining/sector"
+import sealing "github.com/filecoin-project/specs/systems/filecoin_mining/sector"
 
-func (sm *StorageMinerActor_I) SubmitPoSt(postProof base_mining.PoStProof, sectorStateSets sector.SectorStateSets) {
+func (sma *StorageMinerActor_I) SubmitPoSt(postProof base_mining.PoStProof, sectorStateSets sector.SectorStateSets) {
+	panic("TODO")
+}
+
+func (sma *StorageMinerActor_I) CommitSector(onChainInfo sealing.OnChainSealVerifyInfo) {
+	currentSectorContent, found := sma.Sectors()[onChainInfo.SectorNumber()]
+	if found {
+		newSectorContent := append(currentSectorContent, onChainInfo)
+		sma.Sectors()[onChainInfo.SectorNumber()] = newSectorContent
+	} else {
+		sma.Sectors()[onChainInfo.SectorNumber()] = []sealing.OnChainSealVerifyInfo{onChainInfo}
+	}
```

Comment:
Usually, in cases like this, it is idiomatic to handle the exceptions first, then the normal case:
```go
currSector, found := sma.Sectors()[onChainInfo.SectorNumber()]
if !found {
  sma.Sectors()[onChainInfo.SectorNumber()] = []sealing.OnChainSealVerifyInfo{onChainInfo}
} else {
  newSector := append(currSector, onChainInfo)
  sma.Sectors()[onChainInfo.SectorNumber()] = newSector
}

```